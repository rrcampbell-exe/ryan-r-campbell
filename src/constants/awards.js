const awards = [
  { 
    name: '2022 International Book Awards Finalist - Humor and Satire',
    alt: "A wreath containing the words International Book Awards Finalist - 2022 | Humor & Satire",
    link: 'https://www.internationalbookawards.com/2022pressrelease.html',
    src: './awards/2022-international-book-awards-finalist-humor-and-satire.png',
    className: 'light award',
  },
  {
    name: "Readers' Favorite Five Stars",
    alt: "The logo for the Readers' Favorite Five Stars award, which is a silver medallion that features five stars",
    link: 'https://readersfavorite.com/book-review/scambait',
    src: './awards/readers-favorite-five-stars.png',
    className: 'award',
  },
  {
    name: '2020 International Book Awards Finalist - Science Fiction',
    alt: "A wreath containing the words International Book Awards Finalist - 2020 | Science Fiction",
    link: 'https://ryanrcampbell.com/2020/07/23/imminent-dawn-named-as-finalist-in-2020-international-book-awards/',
    src: './awards/2020-international-book-awards-finalist-science-fiction.png',
    className: 'light award',
  }
]

export default awards
