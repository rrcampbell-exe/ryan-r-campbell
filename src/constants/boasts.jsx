import { BrowserSVG, LaptopSVG, ManagementSVG } from '../assets/svg'

const boasts = [
  {
    category: 'Hackathons',
    description: [
      'Finalist in 2022 Hack for Good with augmented reality project to ease treatment for pediatric cancer patients', 
      'Awarded "Most Innovative" in 2024 Developer Experience Hackathon for AI-powered sentiment analysis application'
    ],
    image: <LaptopSVG />
  },
  {
    category: 'Leadership & Culture',
    description: [
      'Founded company-internal podcast to better connect teams and discuss pressing technical topics',
      'Presented on organizational impact of well-crafted mission statements', 
      'Frequent contributor to company blog on topics related to sharpening technical and non-technical skills', 
      'Developed documentation for and trained others on Kubernetes namespace migration'
    ],
    image: <ManagementSVG />
  },
  {
    category: 'UI/UX', 
    description: ['Created React library of reusable components to streamline implementation of common design patterns', 
      'Led team initiative to migrate to modern enterprise design system', 
      'Contributed directly to the codebases of bandwidth-strained teams to squash bugs and improve client experience',
      'Exceeded organizational goal of 60% unit testing coverage by achieving 93.9% and 82.7% coverage for new applications' 
    ],
    image: <BrowserSVG />
  }
]

export default boasts
