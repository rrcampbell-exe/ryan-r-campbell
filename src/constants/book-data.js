const books = [
  {
    title: 'Scambait',
    genre: 'Contemporary Fiction | Humor',
    cover: './books/scambait/scambait-flat-cover-international-book-awards-readers-favorite-sticker.png',
    alt: 'Orange cover of the book Scambait by Ryan R. Campbell, which displays the book title and shows the main character sitting at a desk with a computer and his Saint Bernard at his side',
    summary: 'The International Book Awards finalist in Humor and Satire that asks, "Who scams the scammers?"',
    fullDescription: {
      paragraphOne: 'Ever since his grandma squandered the family fortune to two-bit hucksters from “Microsoft Tech Support,” corporate do-little Eric Amundsen has spent his workdays wasting the time of Nigerian princes and car warranty salesmen everywhere.',
      paragraphTwo: `But when Eric's supposedly long-deceased father contacts him through his spam folder, Eric—whose workday mantra isn't exactly “solutions-oriented”—has serious problems to solve.`
    },
    readMoreLink: './books/scambait/scambait-teaser-chapters.pdf',
    buyLink: 'https://www.amazon.com/dp/B09SY4P6FR',
    shortPromoDescription: 'The 2022 International Book Awards Finalist in Humor and Satire.',
    id: 'scambait'
  },
  {
    title: 'And Ampersand',
    genre: 'Works of Short Fiction',
    cover: './books/and-ampersand/and-ampersand-flat-cover.png',
    alt: 'The black cover of the book And Ampersand by Ryan R. Campbell, which features two large ampersands in gold and white, respectively',
    summary: 'A collection of short fiction that begins with endings, ends with beginnings, and elevates feel over form.',
    fullDescription: {
      paragraphOne: 'Your death comes on slowly, while elsewhere, a nameless man fakes his own. Encased in glass, a young man navigates a world meant for those beyond. A mudslide washes away a once romantic weekend, and a childhood swim drowns all illusion of innocence.',
      paragraphTwo: `These are but some of the tales in this collection of flash fiction, short stories, and freeform prose.`
    },
    readMoreLink: './books/and-ampersand/glassed-excerpt-from-and-ampersand.pdf',
    buyLink: 'https://www.amazon.com/dp/1736387111',
    shortPromoDescription: "The perfect pocket companion for any reader who's tussled with the tie between beginning and end.",
    id: 'and-ampersand'
  },
  {
    title: 'Imminent Dawn',
    genre: 'Science Fiction',
    cover: './books/imminent-dawn/imminent-dawn-flat-cover-international-book-awards-sticker.png',
    alt: 'The blue cover of the book Imminent Dawn by Ryan R. Campbell, which shows the silhouette of a woman against a backdrop of stars and circuitry',
    summary: 'The International Book Awards finalist and first installment in the EMPATHY sci-fi saga.',
    fullDescription: {
      paragraphOne: 'When an art-school dropout enrolls in the first round of human trials for an internet-access brain implant, she never expects to find herself at the center of conspiracies stretching from the highest levels of the North American Union government to the lowest dredges of its shadowy hacking collectives.',
      paragraphTwo: ''
    },
    readMoreLink: './books/imminent-dawn/imminent-dawn-sample-chapter.pdf',
    buyLink: 'https://www.amazon.com/dp/B07N1Q8C6G',
    shortPromoDescription: 'The International Book Awards finalist and first installment in the EMPATHY sci-fi saga.',
    id: 'imminent-dawn'
  },
  {
    title: 'Mourning Dove',
    genre: 'Science Fiction',
    cover: './books/mourning-dove/mourning-dove-flat-cover.png',
    alt: 'The golden cover of the book Mourning Dove by Ryan R. Campbell, which features the silhouettes of a woman with a backpack and a cat set against a technothriller-like web of connected dots',
    summary: 'An evocative, sweeping symphony of love, revenge, and desperation in cacophonous times.',
    fullDescription: {
      paragraphOne: 'In the aftermath of the Human/Etech research study, spouse is further pit against spouse, brother against sister, and governments against their people. In the end, all must choose between reclaiming the past or surrendering to an intractable world of their own creation.',
      paragraphTwo: ''
    },
    readMoreLink: './books/mourning-dove/mourning-dove-sample-chapter.pdf',
    buyLink: 'https://www.amazon.com/dp/B07QZGSDX6',
    shortPromoDescription: 'The sequel to International Book Awards finalist, Imminent Dawn.',
    id: 'mourning-dove'
  },
]

export default books
