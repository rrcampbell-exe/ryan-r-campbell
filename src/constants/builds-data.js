const builds = [
  {
    logo: './builds/real-fake-birds-app-logo.png',
    alt: 'The cartoon outline of a bird against a forest green background',
    title: 'Real Fake Birds',
    teaser: 'Some are real. Some are fake. Tell them apart.',
    description: 'Earn badges and extend your streak in this real-or-fake guessing game that puts even the most ardent birders to the test.',
    technologies: 'React, Node, axios, styled-components, react-router, REST APIs',
    link: 'https://www.realfakebirds.app',
    linkText: 'play the game',
    gitHub: 'https://github.com/rrcampbell-exe/real-fake-birds'
  },
  {
    logo: './builds/yarnsworths-tales-app-logo.png',
    alt: 'The cartoon cat Yarnsworth grins mischieviously',
    title: 'Yarnsworth\'s Tales',
    teaser: 'Trillions of tales await.',
    description: 'This storyline generator for fantasy tabletop roleplaying games inspires players at every stage of gameplay.',
    technologies: 'HTML, JavaScript, CSS, Bootstrap',
    link: 'https://rrcampbell-exe.github.io/yarnsworths-tales/',
    linkText: 'discover your next d&d adventure',
    gitHub: 'https://github.com/rrcampbell-exe/yarnsworths-tales'
  },
  {
    logo: './builds/ryan-r-campbell-app-logo.png',
    alt: 'A black-and-white image of Ryan R. Campbell gazing to the right',
    title: 'The Ryan R. Campbell Website',
    teaser: 'Books and builds and blogs, oh my!',
    description: 'The blog, portfolio, and personal site for author and technologist Ryan R. Campbell.',
    technologies: 'React, Sass, Vite, react-router, visx',
    link: 'https://ryanrcampbell.com/',
    linkText: 'return to the homepage',
    gitHub: 'https://github.com/rrcampbell-exe/ryan-r-campbell'
  },
]

export default builds
