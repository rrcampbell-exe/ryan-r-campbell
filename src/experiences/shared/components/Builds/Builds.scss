.Builds {
  .build {
    display: flex;
    flex-direction: column;
    padding: 2rem;
    border-radius: 4px;
    box-shadow: 0 4px 8px 0 var(--shadow);
    margin-bottom: 2rem;
    transition: 0.3s;
    border: 1px solid var(--main-dark-color);


    &:last-child {
      margin-bottom: 0;
    }

    @media (max-width: 600px) {
      padding: 1rem;
    }
  }

  .app-overview {
    margin-bottom: 0.5rem;
    display: flex;
    flex-direction: column;

    p {
      margin: 0.5rem 0;
    }
  }

  .app-teaser {
    font-size: 0.875rem;
  }

  .app-overview-underline {
    width: 2.5rem;
    height: 2px;
    margin-top: 0.5rem;
  }

  .app-logo {
    border-radius: 4px;
    height: 5rem;
    width: 5rem;

    @media (max-width: 600px) {
      height: 4rem;
      width: 4rem;
    }
  }

  .app-title {
    margin: 0.5rem 0 0 0;
  }

  .app-details {
    p {
      margin: 0.5rem 0;
    }

    &:last-child {
      margin: 0;
    }
  }
}

.dark-theme {
  .build {
    border: 1px solid var(--main-light-color);
  }
}
