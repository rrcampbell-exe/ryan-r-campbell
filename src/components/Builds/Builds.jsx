import React from 'react'
import { builds } from '../../constants'

const Builds = ({ buildsToDisplay }) => {
  const buildsToRender = buildsToDisplay ? builds.slice(1, buildsToDisplay + 1) : builds
  return (
    <div className='Builds'>
      {buildsToRender.map((app) => (
        <div className='build grow-on-hover' key={app.logo} data-testid='build'>
          <div className='app-overview'>
            <img className='app-logo' src={app.logo} alt={app.alt} />
            <h3 className='app-title'>{app.title}</h3>
            <i className='app-teaser'>{app.teaser}</i>
            <p>{app.description}</p>
            <div className='app-overview-underline' />
          </div>
          <div className='app-details'>
            <p><b>Technologies used: </b>{app.technologies}</p>
            <p><a href={app.link} target={app.link && app.link.includes('ryanrcampbell.com') ? '' : '_blank'} rel='noopener'>{app.linkText}</a></p>
            <p><a href={app.gitHub} target='_blank' rel='noopener noreferrer'>view {app.title.toLowerCase()} on github</a></p>
          </div>
        </div>
      ))}
    </div>
  )
}

export default Builds
