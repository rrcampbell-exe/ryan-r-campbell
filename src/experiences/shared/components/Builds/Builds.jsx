import React from 'react'
import { builds } from '../../../../constants'

const Builds = () => {
  return (
    <div className='Builds'>
      {builds.map((app) => (
        <div className='build grow-on-hover'>
          <div className='app-overview'>
            <img className='app-logo' src={app.logo} alt={app.alt} />
            <h3 className='app-title'>{app.title}</h3>
            <i className='app-teaser'>{app.teaser}</i>
            <div className='app-overview-underline' />
          </div>
          <div className='app-details'>
            <p><b>Technologies used: </b>{app.technologies}</p>
            <p><a href={app.link} target='_blank' rel='noopener'>{app.linkText}</a></p>
            <p><a href={app.gitHub} target='_blank' rel='noopener noreferrer'>view {app.title.toLowerCase()} on github</a></p>
          </div>
        </div>
      ))}
    </div>
  )
}

export default Builds
