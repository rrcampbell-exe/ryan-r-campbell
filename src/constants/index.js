import awards from './awards'
import books from './book-data'
import builds from './builds-data'
import featuredIn from './featured-in'
import popText from './pop-text'
import reviews from './reviews'
import shortBio from './short-bio'
import boasts from './boasts.jsx'

export { awards, books, builds, featuredIn, popText, reviews, shortBio, boasts }
