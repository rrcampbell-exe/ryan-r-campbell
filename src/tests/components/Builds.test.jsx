import { describe, expect, test } from 'vitest'
import Builds from '../../experiences/shared/components/Builds/Builds'
import { render } from '@testing-library/react'
import { builds } from '../../constants'

describe('Builds', () => {
  test('default snapshot', () => {
    const { asFragment } = render(<Builds />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render all builds', () => {
    const { getAllByTestId } = render(<Builds />)
    const buildElements = getAllByTestId('build')
    expect(buildElements.length).toBe(builds.length)
  })

  test('should render a subset of builds', () => {
    const { getAllByTestId } = render(<Builds buildsToDisplay={2} />)
    const buildElements = getAllByTestId('build')
    expect(buildElements.length).toBe(2)
  })

})