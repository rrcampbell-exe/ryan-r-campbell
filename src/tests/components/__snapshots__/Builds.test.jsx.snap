// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Builds > default snapshot 1`] = `
<DocumentFragment>
  <div
    class="Builds"
  >
    <div
      class="build grow-on-hover"
      data-testid="build"
    >
      <div
        class="app-overview"
      >
        <img
          alt="A black-and-white image of Ryan R. Campbell gazing to the right"
          class="app-logo"
          src="./builds/ryan-r-campbell-app-logo.png"
        />
        <h3
          class="app-title"
        >
          The Ryan R. Campbell Website
        </h3>
        <i
          class="app-teaser"
        >
          Books and builds and blogs, oh my!
        </i>
        <p>
          The blog, portfolio, and personal site for author and technologist Ryan R. Campbell.
        </p>
        <div
          class="app-overview-underline"
        />
      </div>
      <div
        class="app-details"
      >
        <p>
          <b>
            Technologies used: 
          </b>
          React, Sass, Vite, react-router, visx, Vitest, @testing-library/react, husky
        </p>
        <p>
          <a
            href="https://ryanrcampbell.com/"
            rel="noopener"
            target=""
          >
            return to the homepage
          </a>
        </p>
        <p>
          <a
            href="https://github.com/rrcampbell-exe/ryan-r-campbell"
            rel="noopener noreferrer"
            target="_blank"
          >
            view the ryan r. campbell website on github
          </a>
        </p>
      </div>
    </div>
    <div
      class="build grow-on-hover"
      data-testid="build"
    >
      <div
        class="app-overview"
      >
        <img
          alt="The cartoon outline of a bird against a forest green background"
          class="app-logo"
          src="./builds/real-fake-birds-app-logo.png"
        />
        <h3
          class="app-title"
        >
          Real Fake Birds
        </h3>
        <i
          class="app-teaser"
        >
          Some are real. Some are fake. Tell them apart.
        </i>
        <p>
          Earn badges and extend your streak in this real-or-fake guessing game that puts even the most ardent birders to the test.
        </p>
        <div
          class="app-overview-underline"
        />
      </div>
      <div
        class="app-details"
      >
        <p>
          <b>
            Technologies used: 
          </b>
          React, Node, axios, styled-components, react-router, REST APIs
        </p>
        <p>
          <a
            href="https://www.realfakebirds.app"
            rel="noopener"
            target="_blank"
          >
            play the game
          </a>
        </p>
        <p>
          <a
            href="https://github.com/rrcampbell-exe/real-fake-birds"
            rel="noopener noreferrer"
            target="_blank"
          >
            view real fake birds on github
          </a>
        </p>
      </div>
    </div>
    <div
      class="build grow-on-hover"
      data-testid="build"
    >
      <div
        class="app-overview"
      >
        <img
          alt="The cartoon cat Yarnsworth grins mischieviously"
          class="app-logo"
          src="./builds/yarnsworths-tales-app-logo.png"
        />
        <h3
          class="app-title"
        >
          Yarnsworth's Tales
        </h3>
        <i
          class="app-teaser"
        >
          Trillions of tales await.
        </i>
        <p>
          This storyline generator for fantasy tabletop roleplaying games inspires players at every stage of gameplay.
        </p>
        <div
          class="app-overview-underline"
        />
      </div>
      <div
        class="app-details"
      >
        <p>
          <b>
            Technologies used: 
          </b>
          HTML, JavaScript, CSS, Bootstrap
        </p>
        <p>
          <a
            href="https://rrcampbell-exe.github.io/yarnsworths-tales/"
            rel="noopener"
            target="_blank"
          >
            discover your next d&d adventure
          </a>
        </p>
        <p>
          <a
            href="https://github.com/rrcampbell-exe/yarnsworths-tales"
            rel="noopener noreferrer"
            target="_blank"
          >
            view yarnsworth's tales on github
          </a>
        </p>
      </div>
    </div>
    <div
      class="build grow-on-hover"
      data-testid="build"
    >
      <div
        class="app-overview"
      >
        <img
          alt="A piece of paper with a folded corner and a lower-case letter I"
          class="app-logo"
          src="./builds/file-info-svgrepo-com.png"
        />
        <h3
          class="app-title"
        >
          README Generator
        </h3>
        <i
          class="app-teaser"
        >
          Make READMEs users will actually read.
        </i>
        <p>
          This command line interface README generator takes user-provided input and bundles it tidily for easy consumption. No formatting required!
        </p>
        <div
          class="app-overview-underline"
        />
      </div>
      <div
        class="app-details"
      >
        <p>
          <b>
            Technologies used: 
          </b>
          Node, Inquirer, fs, JavaScript, ESM
        </p>
        <p>
          <a
            rel="noopener"
            target="_blank"
          />
        </p>
        <p>
          <a
            href="https://github.com/rrcampbell-exe/readme-generator"
            rel="noopener noreferrer"
            target="_blank"
          >
            view readme generator on github
          </a>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
